polus-precompute-volume-plugin/src/utils.py

```diff
@@ -193,114 +362,134 @@ def build_pyramid(input_image : str,
+                                                    chunk_size=chunk_size,
+                                                    size=(bf.X, bf.Y, bf.Z),
+                                                    dtype=np.dtype(bf.dtype).name,
+                                                    ids=all_identities,
+                                                    resolution=resolution,
+                                                    segmentation_subdirectory="segment_properties",
+                                                    bit_depth=bit_depth,
+                                                    order="XYZ")
+
+            if imagetype == "image":
+                file_info = nginfo.info_image(directory=output_image,
+                                              dtype=datatype,
+                                              chunk_size = chunk_size,
+                                              size=(bf.X, bf.Y, bf.Z),
+                                              resolution=resolution)
+                    
+            logger.info(f"\n Creating chunked volumes of {chunk_size} based on the info file ...")
+            get_highest_resolution_volumes(bf_image = bf,
+                                           resolution_directory = highest_res_directory)
+
+            logger.info("\n Getting the Rest of the Pyramid ...")
+            for higher_scale in reversed(range(0, num_scales)):
+                # bfshape is XYZ, look at line 357
+                inputshape = np.ceil(np.array(bfshape[:3])/(2**(num_scales-higher_scale-1))).astype('int')
+
+                scale_directory = os.path.join(output_image, str(higher_scale+1)) #images are read from this directory
+                if not os.path.exists(scale_directory):
+                    os.makedirs(scale_directory)
+                assert os.path.exists(scale_directory), f"Key Directory {scale_directory} does not exist"
+                
+                if imagetype == "image":
+                    ngvol.get_rest_of_the_pyramid(directory=scale_directory, input_shape=inputshape, chunk_size=chunk_size,
+                                                datatype=datatype, blurring_method='average')
+                else:
+                    ngvol.get_rest_of_the_pyramid(directory=scale_directory, input_shape=inputshape, chunk_size=chunk_size,
+                                                datatype=datatype, blurring_method='mode')
+                logger.info(f"Saved Encoded Volumes for Scale {higher_scale} from Key Directory {os.path.basename(scale_directory)}")
+
+            logger.info("\n Info basesd on Info File ...")
+            logger.info("Data Type: {}".format(file_info['data_type']))
+            logger.info("Number of Channels: {}".format(file_info['num_channels']))
+            logger.info("Number of Scales: {}".format(len(file_info['scales'])))
+            logger.info("Image Type: {}".format(file_info['type']))
 
     except Exception as e:
```

Comment:
If you're catching a general exception, you should actually reraise the exception so that it halts execution and kills the process.